src/foundry/foundry.js/clientDocuments/chatMessage.d.ts

```diff
@@ -1,16 +1,221 @@
+     * First assume that the currently controlled Token is the speaker
+     *
+     * @param scene - The Scene in which the speaker resides
+     *                (defaultValue: `undefined`)
+     * @param actor - The Actor whom is speaking
+     *                (defaultValue: `undefined`)
+     * @param token - The Token whom is speaking
+     *                (defaultValue: `undefined`)
+     * @param alias - The name of the speaker to display
+     *                (defaultValue: `undefined`)
+     *
+     * @returns The identified speaker data
+     */
+    static getSpeaker({
+      scene,
+      actor,
+      token,
+      alias
+    }?: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      actor?: InstanceType<ConfiguredDocumentClass<typeof Actor>>;
+      token?:
+        | InstanceType<ConfiguredDocumentClass<typeof TokenDocument>>
+        | InstanceType<CONFIG['Token']['objectClass']>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target TokenDocument
+     *
+     * @param token - The TokenDocument of the speaker
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromToken({
+      token,
+      alias
+    }: {
+      token: InstanceType<ConfiguredDocumentClass<typeof BaseToken>> | InstanceType<CONFIG['Token']['objectClass']>;
+      alias: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target Actor
+     *
+     * @param scene - The Scene is which the speaker resides
+     * @param actor - The Actor that is speaking
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromActor({
+      scene,
+      actor,
+      alias
+    }: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      actor?: InstanceType<ConfiguredDocumentClass<typeof Actor>>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * A helper to prepare the speaker object based on a target User
+     *
+     * @param scene - The Scene in which the speaker resides
+     * @param user  - The User who is speaking
+     * @param alias - The name of the speaker to display
+     * @returns The identified speaker data
+     */
+    protected static _getSpeakerFromUser({
+      scene,
+      user,
+      alias
+    }: {
+      scene?: InstanceType<ConfiguredDocumentClass<typeof Scene>>;
+      user?: InstanceType<ConfiguredDocumentClass<typeof User>>;
+      alias?: string;
+    }): ChatSpeakerData;
+
+    /**
+     * Obtain an Actor instance which represents the speaker of this message (if any)
+     * @param speaker - The speaker data object
+     */
+    static getSpeakerActor(speaker: ChatSpeakerData): Actor | null;
```

Comment:
I think this can also work with the _source type. You don't actually need an instance of the `ChatSpeakerData` class. Foundry also uses it that way, e.g. in `TextEditor#_onCLickInlineRoll`, it first calls `ChatMessage#getSpeaker` (which just returns a plain object with ChatSpeakerData source) and then passes it to `ChatMessage#getSpeakerActor`.